test/cypress/integration/04-collective_v2.test.js

```diff
@@ -1,41 +1,86 @@
+  cy.contains('Become a financial contributor');
+  disableSmoothScroll();
+  cy.get(`#section-${section}`).scrollIntoView();
+};
 
+describe('New collective page', () => {
+  let collectiveSlug = null;
   before(() => {
-    cy.createHostedCollective()
+    cy.createHostedCollective({
+      twitterHandle: 'testCollective',
+      githubHandle: 'testCollective',
+      website: 'opencollective.com/testCollective',
+    })
       .then(({ slug }) => (collectiveSlug = slug))
       .then(() => cy.visit(`/${collectiveSlug}/v2`));
   });
 
+  beforeEach(() => {
+    cy.login({ redirect: `/${collectiveSlug}/v2` });
+    cy.wait(900);
+  });
+
   describe('Hero', () => {
-    it.skip('Must have links to twitter, github and website', () => {
-      // TODO
+    it('Must have links to twitter, github and website', () => {
+      cy.get('[data-cy=twitterProfileUrl]').should('have.attr', 'href', 'https://twitter.com/testCollective');
+      cy.get('[data-cy=githubProfileUrl]').should('have.attr', 'href', 'https://github.com/testCollective');
+      cy.get('[data-cy=collectiveWebsite]').should('have.attr', 'href', 'https://opencollective.com/testCollective');
     });
 
-    it.skip('Must have the fiscal host displayed', () => {
-      // TODO
+    it('Must have the fiscal host displayed', () => {
+      cy.get('[data-cy=fiscalHostName]').should('have.visible');
     });
 
-    it.skip('Can change avatar', () => {
-      // TODO
+    it('Can change avatar', () => {
+      uploadImage({
+        dropzone: '[data-cy=heroAvatarDropzone]',
+        file: 'gophercon.jpg',
+      });
+      cy.get('[data-cy=heroAvatarDropzoneSave]').click();
     });
 
-    it.skip('Can edit primary color', () => {
-      // TODO - must check hero or buttons CSS to ensure primary color is properly applied
+    it('Can edit primary color', () => {
+      let color = null;
+
+      cy.get('[data-cy=edit-collective-display-features] [data-cy=edit-main-color-btn]').click({ force: true });
+      cy.get('[data-cy=collective-color-picker-card] [data-cy=collective-color-picker-options-btn]').then($colorBtn => {
+        const randomPick = Math.round(Math.random() * $colorBtn.length);
+        const withFailSafe = $colorBtn[randomPick] || $colorBtn[0];
+
+        cy.wrap(withFailSafe).click();
+        color = withFailSafe.style.backgroundColor;
+      });
+
+      cy.get('[data-cy=collective-color-picker-save-btn]').then($saveBtn => {
+        cy.wrap($saveBtn).should('have.css', 'background-color', color);
+        cy.wrap($saveBtn).click();
+      });
     });
 
-    it.skip('Can change cover background image', () => {
-      // TODO
+    it('Can change cover background image', () => {
+      cy.get('[data-cy=edit-collective-display-features] [data-cy=edit-cover-btn]').click({ force: true });
+      uploadImage({
+        dropzone: '[data-cy=heroBackgroundDropzone]',
+        file: 'gopherBack.png',
+      });
+      cy.get('[data-cy=heroBackgroundDropzoneSave]').click();
```

Comment:
How do we check that the image has properly been updated?

Reply:
I will look into this.